src/base.strategy.ts

```diff
@@ -0,0 +1,46 @@
+import { RestInput } from "./types"
+import { Octokit } from "@octokit/core"
+import { throttling } from "@octokit/plugin-throttling"
+import { getOctokit } from "@actions/github"
+import { GitHub } from "@actions/github/lib/utils"
+import { RequestOptions } from "@octokit/types"
+
+export abstract class BaseStrategy {
```

Comment:
I would like composition over inheritance here, so that each strategy gets an octokit instance passed in as a constructor parameter instead of inheriting and calling the `setupClient` function themselves.

Reply:
I've changed the basestrategy into a non-abstract class, and renamed to PackageClients.
Now the factories instantiate this instance, and pass through the octokit client to the strategies through their constructors.